rs/sns/governance/src/extensions.rs

```diff
@@ -118,21 +123,76 @@ impl ExtensionOperationSpec {
         &self.description
     }
 
-    pub fn validate(&self, arg: ExtensionOperationArg) -> Result<ValidatedOperationArg, String> {
-        (self.validate)(arg)
+    async fn validate_operation_arg(
+        &self,
+        governance: &Governance,
+        arg: ExtensionOperationArg,
+    ) -> Result<ValidatedOperationArg, String> {
+        (self.validate)(governance, arg).await
     }
 }
 
 /// Validates deposit operation arguments
-fn validate_deposit_operation(arg: ExtensionOperationArg) -> Result<ValidatedOperationArg, String> {
-    ValidatedDepositOperationArg::try_from(arg).map(ValidatedOperationArg::TreasuryManagerDeposit)
+fn validate_deposit_operation(
+    governance: &Governance,
+    arg: ExtensionOperationArg,
+) -> BoxFuture<Result<ValidatedOperationArg, String>> {
+    Box::pin(async move {
```

Comment:
Why do we need this?

Reply:
Responded on slack, but summary is b/c every async block is its own type even if they return the same type.